### PR TITLE
Add Auto Release Workflow

### PR DESCRIPTION
**Pull request by naveentatikonda**
### Description
Add Auto Release Workflow
 
### Issues Resolved
https://github.com/opensearch-project/geospatial/issues/219
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/auto-release.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    permissions:
+      contents: write
+    steps:
+      - name: GitHub App token
+        id: github_app_token
+        uses: tibdex/github-app-token@v1.5.0
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+          installation_id: 22958780
```
[the remark this answers is not in the document]
Comment (naveentatikonda) [reply .github/workflows/auto-release.yml:20]: @saratvemulapalli can you please confirm ?

[the remark this answers is not in the document]
Comment (saratvemulapalli) [reply .github/workflows/auto-release.yml:20]: Its same for all workflows within opensearch-project.

File: .github/workflows/auto-release.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Get tag
+        id: tag
+        uses: dawidd6/action-get-tag@v1
+      - uses: actions/checkout@v2
+      - uses: ncipollo/release-action@v1
+        with:
+          github_token: ${{ steps.github_app_token.outputs.token }}
+          bodyFile: release-notes/opensearch-geospatial.release-notes-${{steps.tag.outputs.tag}}.md
```
Comment (saratvemulapalli) [inline .github/workflows/auto-release.yml:28]: Make sure your release notes follow this pattern.

**Review by saratvemulapalli — APPROVED** [review]

**Review by martin-gaievski — APPROVED** [review]

File: .github/workflows/auto-release.yml (hunk 2 above)
Comment (naveentatikonda) [reply .github/workflows/auto-release.yml:28]: Yes, it matches.

**Review by VijayanB — APPROVED** [review]

**Pull request merged**